codebase/src/components/edge.ts

```diff
@@ -1,127 +1,173 @@
-  rad = -rad;
-  let startX = (r - 10) * Math.cos(rad) + x;
-  let startY = (r - 10) * Math.sin(rad) + y;
-
-  const SF = 70;
-
-  let cx1 = startX + SF * Math.cos(rad) + SF * Math.sin(rad);
-  let cy1 = startY + SF * Math.sin(rad) + SF * Math.cos(rad);
-
-  let cx2 = startX + SF * Math.cos(rad) - SF * Math.sin(rad);
-  let cy2 = startY + SF * Math.sin(rad) - SF * Math.cos(rad);
-
-  ctx.beginPath();
-  ctx.lineWidth = 2;
-  ctx.moveTo(startX, startY);
-  ctx.bezierCurveTo(cx1, cy1, cx2, cy2, startX, startY);
-  ctx.stroke();
-
-  ctx.lineWidth = 1;
-  let midPoint = [(cx1 + cx2) / 2 - Math.cos(rad) * 20, (cy1 + cy2) / 2 - Math.sin(rad) * 20];
-
-  ctx.lineWidth = 1;
-  ctx.beginPath();
-  ctx.arc(midPoint[0], midPoint[1], 10, 0, 2 * Math.PI);
-  ctx.fillStyle = "white";
-  ctx.fill();
-  ctx.stroke();
-
-  ctx.font = "15px Arial";
-  ctx.fillStyle = "black";
-  ctx.textBaseline = "middle";
-  ctx.textAlign = "center";
-  ctx.fillText(edge.weight.toString(), midPoint[0], midPoint[1]);
+	rad = -rad;
+	let startX = (r - 10) * Math.cos(rad) + x;
+	let startY = (r - 10) * Math.sin(rad) + y;
+
+	const SF = 70;
+
+	let cx1 = startX + SF * Math.cos(rad) + SF * Math.sin(rad);
+	let cy1 = startY + SF * Math.sin(rad) + SF * Math.cos(rad);
+
+	let cx2 = startX + SF * Math.cos(rad) - SF * Math.sin(rad);
+	let cy2 = startY + SF * Math.sin(rad) - SF * Math.cos(rad);
+
+	ctx.beginPath();
+	ctx.strokeStyle = edge.status;
+	ctx.lineWidth = 2;
+	ctx.moveTo(startX, startY);
+	ctx.bezierCurveTo(cx1, cy1, cx2, cy2, startX, startY);
+	ctx.stroke();
+	ctx.strokeStyle = "black";
+
+	ctx.lineWidth = 1;
+	let midPoint = [(cx1 + cx2) / 2 - Math.cos(rad) * 20, (cy1 + cy2) / 2 - Math.sin(rad) * 20];
+
+	addEdgeWeight(ctx, midPoint[0], midPoint[1], edge.weight);
 }
 
 function drawDirectedEdge(ctx: CanvasRenderingContext2D, fromX: number, fromY: number, toX: number, toY: number, edge: GEdge, circleR: number) {
-  let rad = Math.atan2((fromY - toY), (fromX - toX));
-  let startX = (circleR) * Math.cos(rad) + fromX;
-  let startY = (circleR) * Math.sin(rad) + fromY;
-  let endX = (circleR + 15) * Math.cos(rad) + toX;
-  let endY = (circleR + 15) * Math.sin(rad) + toY;
-  ctx.beginPath();
-  ctx.moveTo(startX, startY);
-  ctx.lineTo(endX, endY);
-  ctx.lineWidth = 3;
-  ctx.stroke();
+	let rad = Math.atan2((fromY - toY), (fromX - toX));
+	let startX = (circleR) * Math.cos(rad) + fromX;
+	let startY = (circleR) * Math.sin(rad) + fromY;
+	let endX = (circleR) * Math.cos(rad) + toX;
+	let endY = (circleR) * Math.sin(rad) + toY;
+
+	let triEndX = (circleR + 15) * Math.cos(rad) + toX;
+	let triEndY = (circleR + 15) * Math.sin(rad) + toY;
+
+	ctx.beginPath();
+	ctx.moveTo(startX, startY);
+	ctx.strokeStyle = edge.status;
+	ctx.lineTo(endX, endY);
+	ctx.lineWidth = 3;
+	ctx.stroke();
+	ctx.strokeStyle = "black";
+
+	let triTipX = (-15) * Math.cos(rad) + triEndX;
+	let triTipY = (-15) * Math.sin(rad) + triEndY;
+	let triLX = (10) * Math.cos(rad + Math.PI / 2) + triEndX;
```

Comment:
if we arent reassigning the variables, declare them as `const` as opposed to `let`